ballista/core/src/config/query.rs

```diff
@@ -0,0 +1,204 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+//! Ballista query configuration
+
+use crate::config::{ConfigEntry, ValidConfiguration, ValidConfigurationBuilder};
+use crate::error::Result;
+use datafusion::arrow::datatypes::DataType;
+use std::collections::HashMap;
+
+pub const BALLISTA_JOB_NAME: &str = "ballista.job.name";
+pub const BALLISTA_DEFAULT_SHUFFLE_PARTITIONS: &str = "ballista.shuffle.partitions";
+pub const BALLISTA_DEFAULT_BATCH_SIZE: &str = "ballista.batch.size";
+pub const BALLISTA_REPARTITION_JOINS: &str = "ballista.repartition.joins";
+pub const BALLISTA_REPARTITION_AGGREGATIONS: &str = "ballista.repartition.aggregations";
+pub const BALLISTA_REPARTITION_WINDOWS: &str = "ballista.repartition.windows";
+pub const BALLISTA_PARQUET_PRUNING: &str = "ballista.parquet.pruning";
+pub const BALLISTA_WITH_INFORMATION_SCHEMA: &str = "ballista.with_information_schema";
+/// give a plugin files dir, and then the dynamic library files in this dir will be load when scheduler state init.
+pub const BALLISTA_PLUGIN_DIR: &str = "ballista.plugin_dir";
+
+/// Ballista configuration, mainly for the query
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct BallistaConfig {
```

Comment:
What is the intent of moving the configs into a `query` namespace? Not all configs will be related to the execution of a single query.

Reply:
The existing configurations seem to be all related to the query execution, which may be used by both the scheduler and executor. The reason to add the query namespace is to distinguish it from the new added SchedulerConfig, since the name `BallistaConfig` may be too general and may give users wrong impression that it includes the SchedulerConfig.